### PR TITLE
Support H7 FMC NOR/PSRAM and SDRAM banks remap or swap

### PR DESCRIPTION
**Pull request by GeorgeCGV**
Adds support for FMC NOR/PSRAM and SDRAM banks remap or swap.

The following values are supported:

* `disabled` - default mapping (reset state).
* `sdram-sram` - swaps the NOR/PSRAM and SDRAM banks.
* `sdramb2` - remaps SDRAM bank 2.

Available only on the H7 series.

### CODE REVIEW

**Comment by erwango** [comment]
^^@[tmilkovic51](https://github.com/tmilkovic51), you might be interested

File: dts/bindings/memory-controllers/st,stm32-fmc.yaml [hunk 1]
```diff
@@ -46,3 +46,20 @@ properties:
...
+        - "sdram-sram"
+        - "sdramb2"
+      description: |
+        The FMC bank mapping configuration (BMAP bits of FMC_BCR1).
+
+        * disable    - Default mapping.
+        * sdram-sram - Swap the NOR/PSRAM bank with SDRAM.
+        * sdramb2    - Remaps the SDRAM bank2.
+
+        Note: Supported only by H7 series.
```
Comment (erwango) [inline dts/bindings/memory-controllers/st,stm32-fmc.yaml:65]: If this is the case, maybe a dedicated `st-stm32h7-fmc` dedicated binding could be added (that would include current binding and add this property).

**Review by erwango — CHANGES_REQUESTED** [review]

File: dts/bindings/memory-controllers/st,stm32-fmc.yaml (hunk 1 above)
Comment (GeorgeCGV) [reply dts/bindings/memory-controllers/st,stm32-fmc.yaml]: Done

**Review by erwango — APPROVED** [review]
LGTM, but I leave @gmarull the last word on this PR

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**